aiida_aurora/workflows/cycling_sequence.py

```diff
@@ -0,0 +1,107 @@
+from aiida import orm
+from aiida.engine import ToContext, WorkChain, append_, while_
+from aiida.plugins import CalculationFactory, DataFactory
+
+CyclerCalcjob = CalculationFactory('aurora.cycler')
+CyclingSpecsData = DataFactory('aurora.cyclingspecs')
+BatterySampleData = DataFactory('aurora.batterysample')
+TomatoSettingsData = DataFactory('aurora.tomatosettings')
+
+
+def validate_inputs(inputs, ctx=None):
+    """Validate the inputs of the entire input namespace."""
+
+    error_message = ''
+    for namekey in inputs['techniques'].keys():
+        if namekey not in inputs['control_settings']:
+            error_message += f'namekey {namekey} missing in control_settings\n'
+
+    for namekey in inputs['control_settings'].keys():
+        if namekey not in inputs['techniques']:
+            error_message += f'namekey {namekey} missing in techniques\n'
+
+    if len(error_message) > 0:
+        return error_message
+
+
+class CyclingSequenceWorkChain(WorkChain):
+    """This workflow represents a process containing a variable number of steps."""
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        # yapf: disable
+        super().define(spec)
+        spec.input(
+            "battery_sample",
+            valid_type=BatterySampleData,
+            help="Battery sample to be used."
+        )
+        spec.input(
+            "tomato_code",
+            valid_type=orm.Code,
+            help="Tomato code to use."
+        )
+
+        spec.input_namespace(
+            "techniques",
```

Comment:
Need to settle on a standard for techniques/protocols through user inquiry. Checking...

Reply:
Received user feedback. This should be `protocols`, which is a collection of individual protocols, each being a sequence of `techniques`.